network-api/networkapi/wagtailpages/pagemodels/base.py

```diff
@@ -597,32 +601,22 @@ class PartnerLogos(WagtailOrderable):
         null=True,
         on_delete=models.SET_NULL,
     )
+    width = models.PositiveSmallIntegerField(
+        help_text='The width of the image. Height will automatically be applied.'
+    )
     panels = [
         ImageChooserPanel('logo'),
         FieldPanel('link'),
+        FieldPanel('width'),
     ]
 
-    class Meta:
-        verbose_name = 'Partner Logo'
-    def clean(self):
-        # Validate internal and external links. Make sure one is always applied
-        # in each Orderable item.
-        super().clean()
-        if self.internal_link and self.external_link:
-            message = "Please only select a page OR enter an external URL"
-            raise ValidationError({
-                'internal_link': ValidationError(message),
-                'external_link': ValidationError(message),
-            })
-        if not self.internal_link and not self.external_link:
-            message = "Please select either page OR enter an external URL"
-            raise ValidationError({
-                'internal_link': ValidationError(message),
-                'external_link': ValidationError(message),
-            })
+    @property
+    def image_rendition(self):
+        return self.logo.get_rendition(f'width-{self.width}')
```

Comment:
I haven't started code review yet. Just saw Tais' comment about image being blurry. I think if we double up `self.width` we can solve the blurry image issue on retina devices.

Reply:
We can add a quality setting to the image, that's my bad I forgot to add that. 

@mmmavis In this case we can't specify 2x width otherwise it'll make the images massive. It's how I'm getting away with different sized images without any special frontend work :/